### ISSUE TITLE
'Show template details' button label and text mismatch when the 'Show button text labels' setting is enabled

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#42371]
### Description

Similar to https://github.com/WordPress/gutenberg/issues/42236 but with an additional issue about translations.

In the Site Editor, when the preference 'Show button text labels' is enabled, the Show template details button shows text instead of the chevron icon:

<img width="1792" alt="Screenshot 2022-07-12 at 15 13 31" src="https://user-images.githubusercontent.com/1682452/178498441-49247bfb-700c-411a-a071-7b9a4cafbfba.png">

The visible text 'Details' is different from the actual accessible name provided by the button aria-label, which is 'Show template details' or 'Show template part details'.

One of the main reasons why the 'Show button text labels' preference was implemented is to help speech recognition software users to use the Editor.

These users need to actually see a control name to be able to voice a command with their speech recognition software. For this to work, the visible text of a control must match (or at least start with) the actual accessible name. See also:

WCAG Success Criterion 2.5.3 Label in Name
https://www.w3.org/TR/WCAG21/#label-in-name

Additionally, part of the aria-label string is not translatable. The `DocumentActions` gets an `entityLabel` prop with hardcoded non-translatable English strings `template` or `template part`:

https://github.com/WordPress/gutenberg/blob/8489bc7416030703c58cdf9f2ee35973dc859523/packages/edit-site/src/components/header/index.js#L179-L188

These strings are used for the Show details button label, to replace the placeholder in a translation function:

`__( 'Show %s details' )`

The resulting string is a mix of translatable and non-translatable words. Instead, a fully translatable string should be provided to translators.

Note: the non-translatable strings are used also for the visually hidden H1 heading on this page. That should be fixed as well.

### Step-by-step reproduction instructions

- Go to the Site Editor
- Enable the preference 'Show button text labels' in Options > Preferences
- Inspect the Show template details button

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
I'd like to propose to simplify as much as possible. I'm not sure the visible text 'Details` is the best choice, to start with. Also, not sure using two aria-labels that distinguish between 'template' and 'template part' adds great value.

Actually, this button opens a panel that provides users with multiple functionalities:
- See the template description.
- Select and Edit template areas, if available.
- Clear customizations.
- Edit the title of custom templates.
- Browse all templates / template parts.

Also, sometimes there are no real 'details' to show (there's no description):

<img width="353" alt="Screenshot 2022-07-12 at 15 44 27" src="https://user-images.githubusercontent.com/1682452/178505400-800535d2-0f25-485d-85fe-7bd7178b8c83.png">

The word 'Details' doesn't seem to clearly identify all these features. It's more a 'manage / browse' templates. I'd like to propose to think of some more appropriate text and use that for both the visible text and the aria-label. That would also solve the issue of the non-translatable parts of the string, at least for the button. The H1 would still need to be fixed.

**Comment by im3dabasia** [comment]
I think the primary objective of this issue can be closed as the labels are now matching the exact text over the button when accessed via a screen reader.

Attaching a screenshot for the same. I believe this is due to the usage of `<DocumentBar/>` in the header component. 

https://github.com/WordPress/gutenberg/blob/f2bd3d2c236fbb089f1dcd11816bae6f3ac45b53/packages/editor/src/components/header/index.js#L151

A change which I believe would improve consistency along with accessibility would be this (based on my observation) Related issue: https://github.com/WordPress/gutenberg/issues/63308

<img width="893" alt="Image" src="https://github.com/user-attachments/assets/ea6e7080-8316-4ad8-9ffb-70e53ec57625" />

Their is a difference in how each of the button has its aria-label attribute setup, while there seems to be a consistent pattern in most of the buttons, 1/2 misses out on this. I think these can be worked on to improve a11y.


| Visual Label | Aria Label           |
|--------------|----------------------|
| Status       | Change status: Draft |
| Publish       | Change date: Immediately |
| Slug       | Change link: details |
| Author       | Change author: admin |
| Template       | Template options |
| Discussion       | Change discussion options |
| Parent       | Change parent: None |


Look majorly there is a pattern "Change x: y", x: being the subject and y being the current value. This pattern is not consistent with Template and Discussion buttons, This is a room for improvement.

Proposed changes

| Visual Label | Aria Label           |
|--------------|----------------------|
| Template       | Change template: options |
| Discussion       | Change discussion: options |

To start of with we can implement the above at-least and progress from there. 'options' can be removed too and we can have the value of the option there. ie 'Change template: page' and 'Change discussion: closed'. 


Looking forward to hear your thoughts.


@t-hamano , @afercia ,  @alexstine

**Comment by t-hamano** [comment]
This issue reports a problem with the document bar, which appears to be resolved in the current UI.

The improvement proposed by @im3dabasia is related to the post summary panel, so it would be a good idea to continue the discussion on #63308.

I'd like to close this issue, but if I've missed something, please feel free to comment.
